app/src/main/java/io/appium/uiautomator2/model/WaitForIdleTimeout.java

```diff
@@ -0,0 +1,19 @@
+package io.appium.uiautomator2.model;
+
+import android.support.test.uiautomator.Configurator;
+
+import io.appium.uiautomator2.utils.Logger;
+
+public class WaitForIdleTimeout {
+
+  public static final String SETTING_NAME = "setWaitForIdleTimeout";
```

Comment:
I would suggest to remove `set` prefix from setting name.

Reply:
Client implementation uses enum constant as "setWaitForIdleTimeout" 
Java_Client:
https://github.com/appium/java-client/blob/37f5ed3677edc68370e98911c7ef89a3c8ee9c1c/src/main/java/io/appium/java_client/Setting.java

appium-dotnet-driver:
https://github.com/appium/appium-dotnet-driver/blob/698351d7e301dc1acb09a5b333907d96686c4281/appium-dotnet-driver/Appium/Android/Enums/AutomatorSetting.cs

Reply:
I don't think it works on client side, because only `ignoreUnimportantViews` implemented in [appium-android-driver](https://github.com/appium/appium-android-driver/blob/master/lib/driver.js#L46-L47) and plus `allowInvisibleElements` in [appium-uiautomator2-driver](https://github.com/appium/appium-uiautomator2-driver/blob/c2133948a084a6d122e1f1448211e9754bcdc3c5/lib/driver.js#L111-L112).
Also you can take a look at additional settings implemented in [appium-android-bootstrap](https://github.com/appium/appium-android-bootstrap/blob/e0df190e35baccf31b4e2af81db75cff580631a1/bootstrap/src/io/appium/android/bootstrap/handler/ConfiguratorHandler.java#L21-L25) but not implemented in driver.